ledger/src/blockstore.rs

```diff
@@ -1484,11 +1487,16 @@ impl Blockstore {
         })
     }
 
+    pub fn get_data_shred(&self, slot: Slot, index: u64) -> Result<Option<Vec<u8>>> {
+        Self::fetch_data_shred(slot, index, &self.data_shred_cf)
+    }
+
     pub fn get_data_shreds_for_slot(
         &self,
         slot: Slot,
         start_index: u64,
     ) -> ShredResult<Vec<Shred>> {
+        use crate::shred::SHRED_PAYLOAD_SIZE;
         self.slot_data_iterator(slot, start_index)
             .expect("blockstore couldn't fetch iterator")
             .map(|data| Shred::new_from_serialized_shred(data.1.to_vec()))
```

Comment:
@carllin - Went back and forth on this for a little bit so figured I'd just open the PR to open up discussion.

The other entrypoints for getting shreds all funnel down nicely to one function. However, this entrypoint that uses an iterator doesn't so we'd potentially have to have repeated logic. Moreover, `slot_data_iterator()` itself is a public function, so someone could use that method directly (and incorrectly).

I'm kind of leaning towards duplicate logic to resize since we know this might be going away soon, but curious if you (or anyone else that happens to see this) have any thoughts on this